filters/include/pcl/filters/impl/local_maximum.hpp

```diff
@@ -146,7 +146,7 @@ pcl::LocalMaximum<PointT>::applyFilterIndices (Indices &indices)
 
     // Check to see if a neighbor is higher than the query point
     float query_z = (*input_)[iii].z;
-    for (std::size_t k = 1; k < radius_indices.size (); ++k)  // k = 1 is the first neighbor
+    for (std::size_t k = 0; k < radius_indices.size (); ++k)  // k = 1 is the first neighbor
```

Comment:
Let's remove the bad comment completely.

Since we are now working with all indices, we can also use a range-for loop:
```cpp
for (const auto& idx: radius_indices) {
  if ((*input_)[idx].z > query.z)  // 1st loop
  point_is_visited[idx] = true;  // 2nd loop
}
```